aces/Loris
# [jsx/Core] Restrict DateElement year to 4 digits

this PR does exactly what this merged PR https://github.com/aces/Loris/pull/3582 did. the code was just removed by a subsequently merged PR (but i won't git blame ;) ) so i am bringing the code back to 20.0. please merge :)

addresses https://redmine.cbrain.mcgill.ca/issues/13767

**to test:**
- on 20.0-release branch: go to Proband Information tab in Candidate Parameters, and see that you can enter a 6 digit year
- on my branch: you are only allowed to enter 4 digits as needed
